File: practice/2022/2/aoc_22_02.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	filePath := "C:/Users/hxzzm/OneDrive/Documents/Git/Advent-of-Code-2023/practice/2022/2/input.txt"
	file, err := os.Open(filePath)
	if err != nil {
		fmt.Println("Error opening file:", err)
		return
	}
	defer file.Close()

	var p1x, p1y int
	p1Map := map[string]int{
		"up":   -1,
		"down": 1,
	}

	var p2x, p2y, p2a int

	scanner := bufio.NewScanner(file)

	for scanner.Scan() {
		line := scanner.Text()
		if len(strings.TrimSpace(line)) == 0 {
			continue
		}

		parts := strings.Split(line, " ")
		s := parts[0]
		intValue, err := strconv.Atoi(parts[1])
		if err != nil {
			fmt.Println("Error parsing integer:", err)
			continue
		}

		if s == "up" || s == "down" {
			p1y += intValue * p1Map[s]
		} else {
			p1x += intValue
		}

		if s == "up" || s == "down" {
			p2a += intValue * p1Map[s]
		} else {
			p2x += intValue
			p2y += p2a * intValue
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	fmt.Println("Part One:", p1y*p1x)
	fmt.Println("Part Two:", p2x*p2y)
}

File: practice/2022/2/aoc_22_02.js
const fs = require("fs");

const filePath =
  "C:/Users/hxzzm/OneDrive/Documents/Git/Advent-of-Code-2023/practice/2022/2/input.txt";
const data = fs.readFileSync(filePath, "utf8");
const lines = data.split("\n").filter(line => line.trim() !== ""); 

// Part One
let p1x = 0
let p1y = 0
const p1Map = {
    'up': -1,
    'down' : 1
}

let s = "", int = 0
lines.forEach(function(line){
    [s, int] = line.split(" ")
    int = parseInt(int)
    if (s === "up" || s === "down"){
        p1y += int * p1Map[s];
    } else {
        p1x += int
    }
});

console.log(p1y * p1x)

// Part Two
let p2x = 0, p2y = 0, p2a = 0;

lines.forEach(function(line){
    [s, int] = line.split(" ")
    int = parseInt(int)
    if (s === "up" || s === "down"){
        p2a += int * p1Map[s];
    } else {
        p2x += int;
        p2y += p2a * int;
    }
});

console.log(p2x * p2y);